### ISSUE TITLE
Support pre-build initialization

### ISSUE THREAD

**Issue by mtomko** [scala-steward-org/scala-steward#2862]
TL;DR: if scala-steward could source a script before running `sbt`, repositories could set up any auth tokens or other things necessary for the build to succeed

Based on discussion [here](https://github.com/broadinstitute/poolq/pull/9#issuecomment-1365804376).

The [poolq](https://github.com/broadinstitute/poolq) build currently uses the [sbt-github-packages](https://github.com/djspiewak/sbt-github-packages) plugin to configure publishing to GitHub packages. That plugin requires some configuration in order to function and it's somewhat awkward in SBT to stub out or delay the initialization of those things, even though they're not needed by `sbt` in order to run the basic tests for the repository. In short, `scala-steward` shouldn`t care but due to limitations in the plugin, it ends up needing to in a way that's rather hard to avoid.

My suggestion is that if `scala-steward` had a way to run some sort of shell script in advance of running `sbt test`, a repository could put in place configuration settings that were necessary for the build to succeed, and putting control in the hands of the repository hopefully would allow the solution to be more general than simply setting a dummy `GITHUB_TOKEN` for repositories that happened to use GitHub packages.

I'd be interested in contributing such a feature, but I wanted to discuss a possible solution here before attempting anything.

**Comment by exoego** [comment]
Since Scala build tools (sbt, mill, etc) supports credential configuration, I think credentials should be configured in build tools (sbt, mill, etc) at user side.
I do not find a good justification for Scala Steward to run shell script just for configuring dummy credential.

If missing credentials affect Scala Steward, I think other tools such as running Scalafix in GitHub Action are also affected.
If credentials is configured properly in build tool, not only Scala Steward but other tools are happy.

**Comment by mkurz** [comment]
+1 
There is the [`postUpdateHooks`](https://github.com/scala-steward-org/scala-steward/blob/main/docs/repo-specific-configuration.md) and IMHO it would be nice to have a `preRunHook` where we could run commands.
